tools/util_darwin.go

```diff
@@ -112,30 +112,11 @@ func CloneFileByPath(dst, src string) (bool, error) {
 }
 
 func cloneFileSyscall(dst, src string) *CloneFileError {
-	srcCString, err := unix.BytePtrFromString(src)
+	err := unix.Clonefileat(C.AT_FDCWD, src, C.AT_FDCWD, dst, C.CLONE_NOFOLLOW)
 	if err != nil {
-		return &CloneFileError{errorString: err.Error()}
-	}
-	dstCString, err := unix.BytePtrFromString(dst)
-	if err != nil {
-		return &CloneFileError{errorString: err.Error()}
-	}
-
-	atFDCwd := C.AT_FDCWD // current directory.
-
-	_, _, errNo := unix.Syscall6(
-		unix.SYS_CLONEFILEAT,
-		uintptr(atFDCwd),
-		uintptr(unsafe.Pointer(srcCString)),
-		uintptr(atFDCwd),
-		uintptr(unsafe.Pointer(dstCString)),
-		uintptr(C.CLONE_NOFOLLOW),
-		0,
-	)
-	if errNo != 0 {
 		return &CloneFileError{
-			Unsupported: errNo == C.ENOTSUP,
-			errorString: fmt.Sprintf("%s. from %v to %v", unix.ErrnoName(errNo), src, dst),
+			Unsupported: err == syscall.ENOTSUP,
+			errorString: fmt.Sprintf("%v from %v to %v: %v", err, src, dst),
```

Comment:
We might be able to a type assertion here with `unix.ErrnoName` here because the syscall takes the `syscall.Errno` value and uses this wrapper:

```go
// errnoErr returns common boxed Errno values, to prevent
// allocations at runtime.
func errnoErr(e syscall.Errno) error {
	switch e {
	case 0:
		return nil
	case EAGAIN:
		return errEAGAIN
	case EINVAL:
		return errEINVAL
	case ENOENT:
		return errENOENT
	}
	return e
}
```

But I'm not quite sure how to test this, so I'm just putting the raw string.

Reply:
It looks like `syscall.Errno` already returns a human-readable string, so we're good here.